File: Go/hello_go/prime.go
package main

import "math"

// isPrime 関数: 指定された数が素数か判定
func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

// getPrimes 関数: n個の素数を求める
func getPrimes(count int) []int {
	primes := []int{}
	num := 2 // 最初の素数
	for len(primes) < count {
		if isPrime(num) {
			primes = append(primes, num)
		}
		num++
	}
	return primes
}

File: JS/hello_hono/my-app/src/prime.ts
// 素数を判定する関数
export function isPrime(num: number): boolean {
  if (num < 2) {
    return false;
  }
  for (let i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      return false;
    }
  }
  return true;
}

// n番目までの素数を列挙する関数
export function getPrime(num: number): number[] {
  const primes = [];
  for (let i = 2; primes.length < num; i++) {
    if (isPrime(i)) {
      primes.push(i);
    }
  }
  return primes;
}